pkg/cmd/cluster/cluster_link.go

```diff
@@ -0,0 +1,129 @@
+package cluster
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"net/url"
+
+	"github.com/dcos/dcos-cli/api"
+	"github.com/dcos/dcos-cli/pkg/config"
+	"github.com/dcos/dcos-cli/pkg/login"
+	"github.com/dcos/dcos-cli/pkg/setup"
+	"github.com/spf13/cobra"
+)
+
+// newCmdClusterLink links the attached cluster to another one.
+func newCmdClusterLink(ctx api.Context) *cobra.Command {
+	setupFlags := setup.NewFlags(ctx.Fs(), ctx.EnvLookup)
+	cmd := &cobra.Command{
+		Use:  "link",
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			attachedCluster, err := ctx.Cluster()
+			if err != nil {
+				return err
+			}
+
+			manager := ctx.ConfigManager()
+			linkableClusterConfig, err := manager.Find(args[0], false)
+			if err != nil {
+				return err
+			}
+
+			if config.Empty() == linkableClusterConfig {
+				// The cluster does not exist yet. Two possibilities:
```

Comment:
If the cluster does not exist, an error would have already been returned at this point. The content of this block should be moved to line 31 instead. We could also add a specific error to make things more robust : 
```
	linkableClusterConfig, err := manager.Find(args[0], false)
	if err != nil {
		if err != config.ErrConfigNotFound {
			return err
		}
		// proceed with cluster setup
	}
```